src/Utils/multi_region_transformation.jl

```diff
@@ -192,9 +192,10 @@ end
 """
     @apply_regionally expr
     
-Use `@apply_regionally` to distribute locally the function calls.
-Call `compute_regionally` in case of a returning value and `apply_regionally!` 
-in case of no return.
+Use `@apply_regionally` to distribute locally the function calls in `expr`ession
+when the functions do not return anything.
+
+In case the function in `expr` returns something, then use [`construct_regionally`](@ref).
```

Comment:
```suggestion
Distributes locally the function calls in `expr`ession

It calls [`apply_regionally!`](@ref) when the functions do not return anything.

In case the function in `expr` returns something, `@apply_regionally` calls [`construct_regionally`](@ref).
```

Reply:
Hm... so I'm confused. Actually the `@apply_regionally` will call `construct_regionally` if the function does not return anything? Is it better to say "fallsback to `construct_regionally` if the function in `expr` returns something."?

Isn't this but prone for bugs?

What will happen with this expression:

```Julia
function foo!(bar)
    bar = 2bar
end
```
versus this:

```Julia
function fooo!(bar)
    bar = 2bar
    return nothing
end
```
?

Reply:
the difference is between
```julia
@apply_regionally ret = returning_function(args)
```
and
```julia
@apply_regionally non_returning_function!(args)
```
the first one calls `construct_regionally` because there is an equal sign, implying a return value. The second is a function call without a return value so it will call `apply_regionally!`